napari/layers/image/_tests/test_multiscale.py

```diff
@@ -441,7 +457,7 @@ def test_multiscale_data_protocol():
         ([[5, 5], [15, 15]], 0, [[5, 5], [15, 15]]),
         # Multiscale level selection uses > rather than >= so use -1 and 21
         # instead of 0 and 20 to ensure that the FOV is big enough.
-        ([[-1, -1], [21, 21]], 1, [[0, 0], [10, 10]]),
+        ([[-1, -1], [21, 21]], 1, [[0, 0], [9, 9]]),
         ([[-11, -11], [31, 31]], 2, [[0, 0], [5, 5]]),
```

Comment:
Since the shape of the lowest resolution level is (5, 5), the corner pixels (which are defined with respect to the displayed data level) can be (4, 4) at most which is what happens in this test.

```suggestion
        ([[-11, -11], [31, 31]], 2, [[0, 0], [4, 4]]),
```

Reply:
But shouldn't the first test also fail with `15 15` and instead require `14 14`?

Reply:
I just need to figure out how the `corner_pixel` are supposed to work; I think the change should actually happen in `ImageSliceRequest`.